sdk/resourcemanager/azure-resourcemanager-appservice/src/main/java/com/azure/resourcemanager/appservice/implementation/WebAppBaseImpl.java

```diff
@@ -214,67 +215,44 @@ private void normalizeProperties() {
         this.diagnosticLogsToUpdate = false;
         this.sslBindingsToCreate = new TreeMap<>();
         this.msiHandler = null;
-        if (inner().hostNames() != null) {
-            this.hostNamesSet = new HashSet<>(inner().hostNames());
-        }
-        if (inner().enabledHostNames() != null) {
-            this.enabledHostNamesSet = new HashSet<>(inner().enabledHostNames());
-        }
-        if (inner().trafficManagerHostNames() != null) {
-            this.trafficManagerHostNamesSet = new HashSet<>(inner().trafficManagerHostNames());
-        }
-        if (inner().outboundIpAddresses() != null) {
-            this.outboundIPAddressesSet = new HashSet<>(Arrays.asList(inner().outboundIpAddresses().split(",[ ]*")));
-        }
-        this.hostNameSslStateMap = new HashMap<>();
-        if (inner().hostnameSslStates() != null) {
-            for (HostnameSslState hostNameSslState : inner().hostnameSslStates()) {
-                // Server returns null sometimes, invalid on update, so we set default
-                if (hostNameSslState.sslState() == null) {
-                    hostNameSslState.withSslState(SslState.DISABLED);
-                }
-                hostNameSslStateMap.put(hostNameSslState.name(), hostNameSslState);
-            }
-        }
+        this.webSiteBase = new WebSiteBaseImpl(inner());
+        this.hostNameSslStateMap = new HashMap<>(this.webSiteBase.hostnameSslStates());
```

Comment:
Why this property remains?

Reply:
In Update flow, code changes the Map to add new items.

Reply:
Then should we just use it to save the update items?

Reply:
Yes. Hence I keep it there.